### PR TITLE
Integrating Fluent2 Divider to main

### PR DESCRIPTION
**Pull request by sophialee0416**
### Platforms Impacted
- [ ] iOS
- [ ] macOS

### Description of changes

This change merges the Fluent2 version of the Divider into main with support for design tokens.
For reference, the changes in the fluent2-tokens branch associated with the Divider are listed below:

- #833 
- #803 

### Verification

| Fluent2 branch                                       | Main                                      |
|----------------------------------------------|--------------------------------------------|
| image | image |

| Before (Separator)                                       | After (MSFDivider)                                      |
|----------------------------------------------|--------------------------------------------|
| image | image |
| image | image |
| image | image |
| image | image |
| image | Simulator Screen Shot - iPhone 12 Pro - 2022-08-17 at 12 50 03 |
| image | image |
| image | image |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1166)

### CODE REVIEW

**Comment by harrieshin** [comment]
how is the tableviewcell landscape bug addressed?

**Comment by harrieshin** [comment]
is the dark / darkelevated color for group/plain style correct for dividers?

**Comment by sophialee0416** [comment]
> how is the tableviewcell landscape bug addressed?

#1167

**Comment by huwilkes** [comment]
Can you do a binary size comparison? I can help with steps if needed.

**Comment by sophialee0416** [comment]
> is the dark / darkelevated color for group/plain style correct for dividers?

Updated the PR description!

**Comment by sophialee0416** [comment]
There was mention in #1167 of slight color changes within the separator->divider: back when the Divider was tokenized there, was a discussion with some designers where it was agreed the Separator was a little too light, so Divider colors were updated (as observed). Do we want to revert this decision and go back to v1 colors?
@huwilkes @harrieshin @edjamesmsft

**Comment by harrieshin** [comment]
> There was mention in #1167 of slight color changes within the separator->divider: back when the Divider was tokenized there, was a discussion with some designers where it was agreed the Separator was a little too light, so Divider colors were updated (as observed). Do we want to revert this decision and go back to v1 colors? @huwilkes @harrieshin @edjamesmsft

I see. so the current divider color is it fluent2 colors ? if it isn't following fluent 1?

**Comment by sophialee0416** [comment]
> > There was mention in #1167 of slight color changes within the separator->divider: back when the Divider was tokenized there, was a discussion with some designers where it was agreed the Separator was a little too light, so Divider colors were updated (as observed). Do we want to revert this decision and go back to v1 colors? @huwilkes @harrieshin @edjamesmsft
> 
> I see. so the current divider color is it fluent2 colors ? if it isn't following fluent 1?

Yes they're from the fluent2 specs!

File: ios/FluentUI/Popup Menu/PopupMenuController.swift [hunk 1]
```diff
@@ -151,7 +152,7 @@ open class PopupMenuController: DrawerController {
         return view
     }()
 
-    private var separator: Separator?
+    private lazy var divider: MSFDivider = .init()
```
Comment (harrieshin) [inline ios/FluentUI/Popup Menu/PopupMenuController.swift:155]: why sometimes we are var foo = MSDivider() and seomtimes var divider: MSFDivider = .init()?

Comment (huwilkes) [reply ios/FluentUI/Popup Menu/PopupMenuController.swift:155]: Good question. I assume I generally tried to match the previous style of the Separator as closely as I could. Though this one is a bit more a change, also going from optional to lazy.

**Review by harrieshin — APPROVED** [review]

File: ios/FluentUI/Popup Menu/PopupMenuController.swift (hunk 1 above)
Comment (mischreiber) [reply ios/FluentUI/Popup Menu/PopupMenuController.swift:155]: One difference could be that member variables always have to be `var divider: MSFDivider` due to a swiftlint rule, while we generally avoid declaring types explicitly inside method bodies.

**Pull request merged**